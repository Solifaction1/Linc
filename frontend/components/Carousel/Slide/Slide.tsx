import React from 'react';

import styles from './slide.module.scss';

type SlideProps = {
    imgUrl: string,
    imgAlt: string
}

const Slide = (props: SlideProps ) => (
    <div className={styles.slideContainer}>
        <img src={props.imgUrl} alt={props.imgAlt || 'Slide'} />
    </div>
);

export default Slide;