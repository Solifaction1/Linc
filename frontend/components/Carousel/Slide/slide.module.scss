.slideContainer {
  position: relative;
  display: flex;
  align-content: center;
  align-items: center;
  justify-content: center;

  width: 100%;
}
