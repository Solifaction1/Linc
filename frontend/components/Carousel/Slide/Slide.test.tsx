import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import Slide from './Slide'

describe('Slide', () => {
	it('renders a Slide', () => {
		const { container } = render(<Slide imgAlt="imgAlt" imgUrl="imgUrl" />)

		expect(container).toMatchSnapshot()

		expect(container.getElementsByClassName('slideContainer')[0]).toBeVisible()
		const img = container.querySelector('img')
		expect(img).toBeVisible()
		expect(img?.getAttribute('alt')).toEqual('imgAlt')
		expect(img?.getAttribute('src')).toEqual('imgUrl')
	})
})
