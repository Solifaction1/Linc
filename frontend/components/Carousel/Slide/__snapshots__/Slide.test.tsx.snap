// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slide renders a Slide 1`] = `
<div>
  <div
    class="slideContainer"
  >
    <img
      alt="imgAlt"
      src="imgUrl"
    />
  </div>
</div>
`;
